src/script/descriptor.cpp

```diff
@@ -536,6 +543,65 @@ std::unique_ptr<Descriptor> ParseScript(Span<const char>& sp, ParseScriptContext
     return nullptr;
 }
 
+
+std::unique_ptr<PubkeyProvider> InferPubkey(const CPubKey& pubkey, ParseScriptContext, const SigningProvider& provider)
+{
+    std::unique_ptr<PubkeyProvider> key_provider = MakeUnique<ConstPubkeyProvider>(pubkey);
+    return key_provider;
```

Comment:
Noting differences here: https://github.com/bitcoin/bitcoin/pull/14477/files#diff-dcb25ae6b5207f65ae6a1cd364c9e62e5a9df69a1cd21b9ed79849cc99c2150bR638-R641

@UdjinM6 are these important?

Reply:
I guess so. I don't see why would we want to drop these lines.

Reply:
Looks like this hasn't been resolved?

Reply:
It looks like OriginPubkeyProvider was introduced in bitcoin PR #14150.
Currently blocked by backport #14150 which is not merged yet.

Reply:
Okay, so we (presumably) need to backport 14150 then. Would you like to do this?

Reply:
It is currently under dev by vijaydasmp

Reply:
yes @ogabrielides its part of https://github.com/dashpay/dash/pull/4430 , yesterday I faced system failure messages from CI system, I will retry today

Reply:
@vijaydasmp any progress on that?

Reply:
Hello @ogabrielides,  I had to drop that commit from #4430 due to dependency on another complex merge, however, I tried it separately #4457 after I saw your message to unblock your work, but I am not able to fix the errors quickly yet, please feel free to take it and use it in your pr, I will drop it